### PR TITLE
Add trigger file creation support, useful if using tesla_dashcam to combine cam videos.

### PR DESCRIPTION
**Pull request by croadfeldt**

### CODE REVIEW

**Comment by croadfeldt** [comment]
ShellCheck results are invalid, see link for reason why.
[https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash](https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash)

**Comment by marcone** [comment]
> ShellCheck results are invalid, see link for reason why.
> https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

From that link: "while quotes can be safely omitted  [...] it would sometimes be better to write the solution with quotes as [ -z "${var+x}" ]"
I think of shellcheck as compiler warnings: they may not *always* point out a real problem, but they do so often enough that it's worth tweaking the code to avoid the warnings.

File: run/archiveloop [hunk 1]
```diff
@@ -308,6 +308,12 @@ function archive_teslacam_clips () {
             fi
         fi
 
+	# Create trigger file for SavedClips
+	if [ ! -z ${ARCHIVE_TRIGGER_SAVED+x} ] ; then export ARCHIVE_TRIGGER_SAVED; fi
```
Comment (marcone) [inline run/archiveloop]: please if/then style used in the rest of the scripts:
```
if condition
then
  commands
fi
```

File: run/cifs_archive/archive-clips.sh [hunk 2]
```diff
@@ -81,9 +81,15 @@ connectionmonitor $$ &
...
 
+# Create trigger file for SavedClips
+if [ ! -z ${ARCHIVE_TRIGGER_SAVED+x} ] ; then log "Creating SavedClips Trigger File: $ARCHIVE_MOUNT/SavedClips/${ARCHIVE_TRIGGER_SAVED}"; touch "$ARCHIVE_MOUNT/SavedClips/${ARCHIVE_TRIGGER_SAVED}"; fi
+
 # v10 firmware adds a SentryClips folder
 moveclips "$CAM_MOUNT/TeslaCam/SentryClips"
 
+# Create trigger file for SentryClips
+if [ ! -z ${ARCHIVE_TRIGGER_SENTRY+x} ] ; then log "Creating SentryClips Trigger File: $ARCHIVE_MOUNT/SentryClips/${ARCHIVE_TRIGGER_SENTRY}"; touch "$ARCHIVE_MOUNT/SentryClips/${ARCHIVE_TRIGGER_SENTRY}"; fi
+
```
Comment (marcone) [inline run/cifs_archive/archive-clips.sh:100]: use if/then style used in rest of scripts, and put each command on a separate line

**Comment by croadfeldt** [comment]
Matched the existing coding style as requested. Assuming folks will be avoiding spaces and special characters in trigger file name. If you'd like, can add support for that as well.

**Comment by marcone** [comment]
Shellcheck still doesn't like it. You could either use uppercase variable names (see https://github.com/koalaman/shellcheck/wiki/SC2154) or explicitly disable the warning using `# shellcheck disable=SC2154` (this is already done in some other places in the scripts)

Also, please squash all changes together and upload as a single change (i.e. use `git rebase -i` and `git push --force`)

**Comment by croadfeldt** [comment]
Let me know if you'd like any other changes.

File: pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample [hunk 3]
```diff
@@ -118,6 +118,18 @@ export HEADLESS_SETUP=true
 # export aws_secret_key=put_your_secretkey_here
 # export aws_sns_topic_arn=put_your_sns_topicarn_here
 
+# tesla_dashcam / Trigger file support.
+# Uncomment any trigger_file_xxxxx below if you'd like to create trigger file(s) in the 
+# archive directory for the listed clip type after all clips have been transferred for that type.
+# eg; export trigger_file_saved=ARCHIVE_UPLOADED would create a file in $ARCHIVE_BASE/SavedClips/ARCHIVE_UPLOADED on your CIFS / Samba server.
+# This is usefull if using https://github.com/ehendrix23/tesla_dashcam to further process clips.
```
Comment (marcone) [inline pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample]: typo: "usefull"

File: pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample [hunk 4]
```diff
@@ -118,6 +118,18 @@ export HEADLESS_SETUP=true
...
+# eg; tesla_dashcom --monitor --monitor_trigger $ARCHIVE_BASE/SavedClips/ARCHIVE_UPLOADED $ARCHIVE_BASE/SavedClips
+#
+# You may modify the value to your desired trigger filename, if you so choose. Remember to alter you tesla_dashcam trigger filename to match.
```
Comment (marcone) [inline pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample]: typo: "you" in 2nd sentence should be "your"

**Review by marcone — APPROVED** [review]
Noticed two typos, but looks OK otherwise.

**Comment by croadfeldt** [comment]
Let me go fix those quick... :)

**Comment by croadfeldt** [comment]
Done.

**Comment by withinfocus** [comment]
@croadfeldt do you have any documentation on your setup using these? While I am concerned about the Pi's ability to do all the `ffmpeg` work I have been wanting to set up a similar workflow where I process the clips and then `rclone` them somewhere.

**Comment by croadfeldt** [comment]
My setup is below.

File storage / processing: Linux server, 9700K, using ZFS for storage.
Pi: PI Zero Wireless
SD Card: 256GB Samsung EVO, will see on reliability before moving to a HE card.
CIFS docker: dperson/samba:latest

I created a ZFS dataset for initial upload by teslausb via CIFS/SMB. the CIFS/SMB server is a docker image to isolate it solely for teslausb. The dedicated ZFS dataset is mapped via podman / docker to be the only storage available for this CIFS/SMB instance.

When the teslausb does it's detection routine and completes the uploads of SavedClips or SentryClips, it also adds the trigger file in the base directory for either.

On the linux server itself, I have tesla_dashcam running natively, soon to be isolated in a container though. There are two instances of tesla_dashcam running, both via a single tmux session so I can monitor them and quasi daemonize them. These instances of tesla_dashcam each monitor one of the target directories of the CIFS/SMB instance at the file system level, since they are running natively and have direct access to the ZFS dataset. Once I move it to a container, I will map that directory into the container along with the output directories. When the respective dashcam instance detects the presence of the trigger file, it does it's thing, searches for new files to merge together. The output of tesla_dashcam go into additional ZFS datasets which are mounted to directories which I have plex setup to share.

From start to finish, it usually takes about 5-10 miniutes after a WIFI connection for each dashcam / sentry clip to be processed and available via plex. This all happens automatically using teslausb's automated file upload and tesla_dashcam trigger file support now. It could be done without the trigger file support easy enough if desired, just need to have the tesla_dashcam instances loop every 30-60 seconds or so.

Re: Pi doing all the work. It should be possible on a pi4, using a cpu cooler. For me it was easier, faster and more robust to use my homelab infrastructure to do all the heavy lifting.

Hope this helps, if you want a more detailed write up, let me know. It's a fairly safe and extremely convenient setup.

**Pull request merged**